src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        (i === this.length - 1)
+          ? result = result + this[i]
+          : result = result + this[i] + separator;
+      } else {
+        (i === this.length - 1)
+          ? result += ''
+          : result += '' + separator;
```

Comment:
do you really need to add an empty line to the result if the current element is the last one?